Frontend/src/pages/register/components/RegistrationPanel.jsx

```diff
@@ -1,48 +1,141 @@
+    }
+  }, [registrationRequest])
+  const { mutate: updateRegistrationMutation } = useMutation({
+    mutationFn: updateRegistration,
+    onError: (data) => {
+      setMessage(
+        'Registration update failed with error: ' + data.error,
+        'negative'
+      )
+    },
+    onSuccess: (data) => {
+      setMessage('Registration update succeeded', 'positive')
+      queryClient.setQueryData(['registration', competitionInfo.id, user], data)
+    },
+  })
+  const { mutate: createRegistrationMutation } = useMutation({
+    mutationFn: submitEventRegistration,
+    onError: (data) => {
+      setMessage('Registration failed with error: ' + data.error, 'negative')
+    },
+    onSuccess: (_) => {
+      // We can't update the registration yet, because there might be more steps needed
+      // And the Registration might still be processing
+      setMessage('Registration submitted successfully', 'positive')
+    },
+  })
+
+  return isLoading ? (
+    <LoadingMessage />
+  ) : (
     <div className={styles.panel}>
-      <EventSelector
-        handleEventSelection={setSelectedEvents}
-        events={competitionInfo.event_ids}
-        initialSelected={[]}
-        size="2x"
+      {registration.registration_status ? (
+        <>
+          <h3>You have registered for {competitionInfo.name}</h3>
+          {/* Really weird bug here: Only if I have the console.log statement here (or any other statement referencing it), initialSelected, will work correctly?? */}
+          {/* eslint-disable-next-line no-console */}
+          {console.log(registration.event_ids)}
+          <EventSelector
+            handleEventSelection={setSelectedEvents}
+            events={competitionInfo.event_ids}
+            initialSelected={registration.event_ids}
+            size="2x"
+          />
+        </>
+      ) : (
+        <>
+          <h3> You can register for {competitionInfo.name}</h3>
+          <EventSelector
+            handleEventSelection={setSelectedEvents}
+            events={competitionInfo.event_ids}
+            initialSelected={[]}
+            size="2x"
+          />
+        </>
+      )}
+
+      <TextArea
+        onChange={(_, data) => setComment(data.value)}
+        value={comment}
       />
-      <TextArea onChange={(_, data) => setComment(data.value)}> </TextArea>
-      <Button
-        onClick={async () => {
-          setMessage('Registration is being processed', 'basic')
-          const response = await submitEventRegistration(
-            user,
-            competitionInfo.id,
-            comment,
-            selectedEvents
-          )
-          if (response.error) {
-            setMessage(
-              'Registration failed with error: ' + response.error,
-              'negative'
-            )
-          } else {
-            setMessage('Registration succeeded', 'positive')
-          }
-        }}
-        positive
-      >
-        Register
-      </Button>
+      {registration.registration_status ? (
```

Comment:
These buttons should probably be disabled while the query or the respective mutations are loading?

Reply:
good point